src/binary.ts

```diff
@@ -1,36 +1,110 @@
-const { BinarySerializer, BytesList } = require("./serdes/binary-serializer");
-const { bytesToHex, slice, parseBytes } = require("./utils/bytes-utils");
+import { BinaryParser } from "./serdes/binary-parser";
+import { AccountID } from "./types/account-id";
+import { HashPrefix } from "./hash-prefixes";
+import { BinarySerializer, BytesList } from "./serdes/binary-serializer";
+import { sha512Half, transactionID } from "./hashes";
+import { FieldInstance } from "./enums";
+import { STObject } from "./types/st-object";
 
-const { sha512Half, transactionID } = require("./hashes");
+/**
+ * Construct a BinaryParser
+ *
+ * @param bytes hex-string to construct BinaryParser from
+ * @returns A BinaryParser
+ */
+const makeParser = (bytes: string): BinaryParser => new BinaryParser(bytes);
 
-const makeParser = (bytes) => new BinaryParser(bytes);
-const readJSON = (parser) => parser.readType(coreTypes.STObject).toJSON();
-const binaryToJSON = (bytes) => readJSON(makeParser(bytes));
+/**
+ * Parse BinaryParser into JSON
+ *
+ * @param parser BinaryParser object
+ * @returns JSON for the bytes in the BinaryParser
+ */
+const readJSON = (parser: BinaryParser): Record<string, unknown> =>
+  (parser.readType(coreTypes.STObject) as STObject).toJSON();
 
-function serializeObject(object, opts = <any>{}) {
+/**
+ * Parse a hex-string into its JSON interpretation
+ *
+ * @param bytes hex-string to parse into JSON
+ * @returns JSON
+ */
+const binaryToJSON = (bytes: string): Record<string, unknown> =>
+  readJSON(makeParser(bytes));
+
+/**
+ * Interface for passing parameters to SerializeObject
+ *
+ * @field set signingFieldOnly to true if you want to serialize only signing fields
+ */
+interface OptionObject {
+  prefix?: Buffer;
+  suffix?: Buffer;
+  signingFieldsOnly?: boolean;
+}
+
+/**
+ * Function to serialize JSON object representing a transaction
+ *
+ * @param object JSON object to serialize
+ * @param opts options for serializing, including optional prefix, suffix, and signingFieldOnly
+ * @returns A Buffer containing the serialized object
+ */
+function serializeObject(
+  object: Record<string, unknown>,
+  opts: OptionObject = {}
+): Buffer {
   const { prefix, suffix, signingFieldsOnly = false } = opts;
   const bytesList = new BytesList();
+
   if (prefix) {
     bytesList.put(prefix);
   }
-  const filter = signingFieldsOnly ? (f) => f.isSigningField : undefined;
+
+  const filter = signingFieldsOnly
+    ? (f: FieldInstance): boolean => f.isSigningField
+    : undefined;
   coreTypes.STObject.from(object, filter).toBytesSink(bytesList);
+
   if (suffix) {
     bytesList.put(suffix);
   }
+
   return bytesList.toBytes();
 }
 
-function signingData(tx, prefix = HashPrefix.transactionSig) {
+/**
+ * Serialize an object for signing
+ *
+ * @param tx Transaction to serialize
+ * @param prefix Prefix bytes to put before the serialized object
+ * @returns A Buffer with the serialized object
+ */
+function signingData(
+  tx: Record<string, unknown>,
```

Comment:
recommend `transaction` which I think we've been using other places for consistency